Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -2357,6 +2355,53 @@ void ProcessMolProps(RWMol *mol) {
       }
     }
   }
+  for (auto &sg : getSubstanceGroups(*mol)) {
+    if (sg.getProp<std::string>("TYPE") == "DAT") {
+      std::string fieldn;
+      if (sg.getPropIfPresent("FIELDNAME", fieldn) &&
+          fieldn == "MRV_IMPLICIT_H") {
+        // CXN extension to specify implicit Hs, used for aromatic rings
+        std::vector<std::string> dataFields;
+        if (sg.getPropIfPresent("DATAFIELDS", dataFields)) {
+          for (const auto &df : dataFields) {
+            if (df.substr(0, 6) == "IMPL_H") {
+              auto val = FileParserUtils::toInt(df.substr(6));
+              for (auto atIdx : sg.getAtoms()) {
+                if (atIdx < mol->getNumAtoms()) {
+                  // if the atom has aromatic bonds to it, then set the explicit
+                  // value, otherwise skip it.
+                  auto atom = mol->getAtomWithIdx(atIdx);
+                  bool hasAromaticBonds = false;
+                  for (auto bndI :
+                       boost::make_iterator_range(mol->getAtomBonds(atom))) {
+                    auto bnd = (*mol)[bndI];
+                    if (bnd->getIsAromatic() ||
+                        bnd->getBondType() == Bond::AROMATIC) {
+                      hasAromaticBonds = true;
+                      break;
+                    }
+                  }
+                  if (hasAromaticBonds) {
+                    atom->setNumExplicitHs(val);
```

Comment:
It seems like this provides an idiomatic RDKit way to understand this specific data from an SD file. I think it might make sense to remove the SGroup after its data has been successfully turned the explicitH property.

Reply:
Makes sense. Since we are processing the data we might as well remove the SGroup. I’ll do that later today or tomorrow morning

Reply:
Makes sense. Since we are processing the data we might as well remove the SGroup. I’ll do that later today or tomorrow morning

Reply:
Makes sense. Since we are processing the data we might as well remove the SGroup. I’ll do that later today or tomorrow morning